nvflare/apis/impl/wf_comm_client.py

```diff
@@ -57,7 +59,6 @@ def broadcast_and_wait(
         abort_signal: Signal = None,
     ):
         engine = fl_ctx.get_engine()
-        request = task.data
         # apply task filters
```

Comment:
if we do the old way of `request = task.data` what is the problem of it?

Reply:
This is because the `before_task_sent_cb` of CrossSiteModelEval sets the `task.data` to new data rather than modifying the original data, so `request` will still point to the old data. To fix this I moved `request = task.data` to right before `send_aux_request()` so it will point to the correct new `task.data`.

https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_common/workflows/cross_site_model_eval.py#L262

Reply:
got it, let's add a comment saying "we need to get this task.data since the user callback might change it"